custom/rch/models.py

```diff
@@ -0,0 +1,159 @@
+        return self._get_rch_id_key(self.doc_type)
+
+    def _set_mother_fields(self, dict_of_props):
+        self.aadhar_num = dict_of_props['PW_Aadhar_No']
+        self.name = dict_of_props['Name_wife']
+        self.dob = dict_of_props['Mother_BirthDate']
+
+    def _set_child_fields(self, dict_of_props):
+        self.aadhar_num = dict_of_props['Child_Aadhaar_No']
+        self.name = dict_of_props['Name_Child']
+        self.dob = dict_of_props['Birth_Date']
+
+    def _set_custom_fields(self, dict_of_props):
+        if self.mother_record():
+            self._set_mother_fields(dict_of_props)
+        elif self.child_record():
+            self._set_child_fields(dict_of_props)
+
+    def assign_search_fields(self, dict_of_props):
+        # set fields from details received from RCH that are specifically used for quick lookup for
+        # filtering or matching
+        self.district_id = dict_of_props['MDDS_DistrictID']
+        self.state_id = dict_of_props['MDDS_StateID']
+        self.village_id = dict_of_props['MDDS_VillageID']
+        self.rch_id = dict_of_props[self.rch_id_key]
+        self.village_name = dict_of_props.get('Village_Name')
+        self._set_custom_fields(dict_of_props)
+
+    @classmethod
+    def update_beneficiaries(cls, beneficiary_type, days_before=1):
+        date_str = str(datetime.date.fromordinal(datetime.date.today().toordinal() - days_before))
+        for state_id in STATE_DISTRICT_MAPPING:
+            for district_id in STATE_DISTRICT_MAPPING[state_id]:
+                records = fetch_beneficiaries_records(date_str, state_id, beneficiary_type, district_id)
+                for record in records:
+                    # convert list of dicts of properties to a single dict
+                    dict_of_props = {}
+                    for prop in record:
+                        if prop.keys()[0] in cls.accepted_fields(beneficiary_type):
+                            dict_of_props[prop.keys()[0]] = prop.values()[0]
+
+                    rch_id_key_field = cls._get_rch_id_key(beneficiary_type)
+                    record_pk = dict_of_props[rch_id_key_field]
+                    assert record_pk
+                    # Find corresponding record if already present using RCH ID
+                    rch_beneficiary = cls.objects.filter(rch_id=record_pk).first()
+                    # else initialize if new record to be added
+                    if not rch_beneficiary:
+                        rch_beneficiary = cls(doc_type=beneficiary_type)
+
+                    rch_beneficiary.assign_search_fields(dict_of_props)
+                    rch_beneficiary.details = dict_of_props
+                    rch_beneficiary.save()
+
+    def associate_matching_cas_record(self):
+        matching_icds_cas_case_id = find_matching_cas_record_id(self.aadhar_num)
+        if matching_icds_cas_case_id:
+            self.cas_case_id = matching_icds_cas_case_id
+            self.save()
+
+
+class AreaMapping(models.Model):
```

Comment:
might need some indexes on some of these fields depending on how big the table will be

Reply:
As of now this table has 873 rows but would grow. Adding indexes.